tests/unit/spanner_dbapi/test_connection.py

```diff
@@ -239,20 +211,20 @@ def test_snapshot_checkout(self):
         self.assertEqual(snapshot, connection.snapshot_checkout())
 
         connection.commit()
-        self.assertIsNone(connection._snapshot)
+        self.assertIsNotNone(connection._snapshot)
```

Comment:
(Not related to this line, but it made me think about the following situation) What happens if the user executes the following?

```
autocommit = true;

select * from my_table;
show variable read_timestamp; -- This should return a read timestamp

begin;
insert into my_table (id) values (1);
commit;
show variable commit_timestamp; -- this should show a commit timestamp
show variable read_timestamp; -- this should return nothing
```

My basic question is: Do we remove the `connection._snapshot` when a read/write transaction is started? Or could it be that we have both a `_snapshot` and a `_transaction` on the connection?

Reply:
Thanks!